packages/mobile/src/components/header-button/HeaderButton.tsx

```diff
@@ -0,0 +1,48 @@
+import { ViewStyle } from 'react-native'
+
+import Button from 'app/components/button'
+import { font, makeStyles, WithStyles } from 'app/styles'
+import { spacing } from 'app/styles/spacing'
+import { useThemeColors } from 'app/utils/theme'
+
+type HeaderButtonProps = WithStyles<
```

Comment:
I like the full with styles wrapper, one other option is that we intersect it? feeling that might keep things a bit more modular?

Reply:
You're right, this is so much better: 
```ts
type HeaderButtonProps = {
  onPress: () => void
  text: string
} & StylesProps<{ root: ViewStyle }>
```
Now we do have `StyleProp`, `StylesProp`, and `StylesProps` though lol

And I just forgot to use the `styles` prop 😅  fixing now

Reply:
that is pretty funny, i guess we just have stylesprop as the only exported thing then ultimately?

Reply:
Yup, eventually we can have everything following the `StylesProps` pattern